### PR TITLE
Fix Unnerve Activation Timing

### PR DESCRIPTION
**Pull request by JFernandez-CC**
Added event "TrySwitchIn" for unnerve timing to allow it to activate
before hazards. Tested with normal switch ins and drag outs.

### CODE REVIEW

**Comment by Zarel** [comment]
Are you sure you got the order correct? Unnerve on both Pokémon should activate _before_ things like Stealth Rock damage.

**Comment by JFernandez-CC** [comment]
Hm, you're right, I forgot to test using two Unnerve Pokemon.

File: battle-engine.js [hunk 1]
```diff
@@ -4093,6 +4094,7 @@ Battle = (() => {
 			let pokemon = this.p1.active[i];
 			if (pokemon.forceSwitchFlag) {
 				if (pokemon.hp) this.dragIn(pokemon.side, pokemon.position);
+				runEvent('TrySwitchIn', pokemon);
```
Comment (ascriptmaster) [inline battle-engine.js]: `this.runEvent`

Comment (Zarel) [inline battle-engine.js]: That's a good point, there's no way this _doesn't_ crash...

**Comment by Zarel** [comment]
Since this code doesn't actually fix the bug it claims to fix, and has gone unmaintained for nearly two weeks, I'm closing it. Feel free to reopen if you do have plans to fix this.

**Pull request closed without merging**